### ISSUE TITLE
`unnecessary-literal-within-tuple-call` (C409) rule transforms single element list literals incorrectly

### ISSUE THREAD

**Issue by bardiharborow** [astral-sh/ruff#10323]
The `unnecessary-literal-within-tuple-call` ([C409](https://docs.astral.sh/ruff/rules/unnecessary-literal-within-tuple-call/)) rule detects the violation here:
```py
tuple([1])
```

`ruff check --fix --unsafe-fixes` fails to transform the code correctly, resulting in this, which is not a tuple:
```py
(1)
```

Instead it should transform the code to:
```py
(1,)
```

Environment:

```shell
% ruff --version
ruff 0.3.2

% cat pyproject.toml
[tool.ruff]
lint.select = [
    "C4"
]
```

**Comment by WindowGenerator** [comment]
@zanieb Hi!
Can I take it too?

**Comment by charliermarsh** [comment]
Go for it @WindowGenerator.

**Comment by bardiharborow** [comment]
I don't know much about Ruff internals but it feels like it might be safer to take the approach of transforming and then rendering the AST, rather than replacing specific string indexes, since the latter is vulnerable to producing invalid or incorrect code in this fashion.

**Comment by dhruvmanila** [comment]
Yeah, we have a [`Generator`](https://github.com/astral-sh/ruff/blob/2cad4bbb354f5d43131b2d9cd583caad8e65e840/crates/ruff_python_codegen/src/generator.rs#L64) struct which does exactly that. Moving the fix to use that would solve the issue as it takes one element tuple into consideration.

**Comment by charliermarsh** [comment]
We should just add this case to the fix. Text-based fixes are preferable because they preserve all trivia. (This fix used to use LibCST but we intentionally changed it because it’s slow and we want to remove our LibCST dependency altogether eventually.)

I think there was just a misunderstanding on what the fix is supposed to do based on the comment above it in the code, which led to this oversight.

**Comment by bardiharborow** [comment]
Fixed by #10491.
